svf/lib/Util/ExtAPI.cpp

```diff
@@ -80,55 +88,77 @@ static std::string GetStdoutFromCommand(const std::string& command)
 // Get extapi.bc file path in npm
 static std::string getFilePath(const std::string& path)
 {
-    std::string bcFilePath = GetStdoutFromCommand(path);
-    if (path.compare("npm root") == 0)
+    std::string bcFilePath = "";
+    if (path.compare("SVF_DIR") == 0)
     {
-        int os_flag = 1;
+        bcFilePath = getenv("SVF_DIR");
+    }
+    else if (path.compare("npm root") == 0)
+    {
+        bcFilePath = GetStdoutFromCommand(path);
         // SVF installed via npm needs to determine the type of operating
         // system, otherwise the extapi.bc path may not be found.
 #ifdef linux
         // Linux os
-        os_flag = 0;
         bcFilePath.append("/svf-lib/SVF-linux");
-#endif
+#else
         // Mac os
-        if (os_flag == 1)
-        {
-            bcFilePath.append("/svf-lib/SVF-osx");
-        }
+        bcFilePath.append("/svf-lib/SVF-osx");
+#endif
     }
 
-    if (bcFilePath.back() != '/')
+    if (!bcFilePath.empty() && bcFilePath.back() != '/')
         bcFilePath.push_back('/');
-    bcFilePath.append(EXTAPI_BC_PATH);
+    bcFilePath.append(BUILD_TYPE).append(DEFUALT_EXTAPI_BC_PATH);
     return bcFilePath;
 }
 
 // Get extapi.bc path
 std::string ExtAPI::getExtBcPath()
 {
-    struct stat statbuf;
-    std::string bcFilePath = std::string(EXTAPI_DIR) + "/extapi.bc";
-    if (!stat(bcFilePath.c_str(), &statbuf))
-        return bcFilePath;
-
-    bcFilePath = getFilePath("echo $SVF_DIR");
-    if (!stat(bcFilePath.c_str(), &statbuf))
-        return bcFilePath;
+    // Five default ways to get extapi.bc path
+    // 1. Set extapi.bc path through setExtBcPath() method
+    // 2. Set extapi.bc path through the "-extapi = PATH_TO_EXTAPIFILE" option
+    // 3. Default path: "SVF_IR_PATH/CMAKE_BUILD_TYPE-build/svf-llvm/extapi.bc"
+    // 4. From $SVF_DIR
+    // 5. From "npm root"(If SVF is installed via npm)
 
-    bcFilePath = getFilePath("npm root");
-    if (!stat(bcFilePath.c_str(), &statbuf))
-        return bcFilePath;
-
-    SVFUtil::errs() << "No extapi.bc found at " << bcFilePath << " for getExtAPI(); The default path for extapi.bc is: SVF_IR_PATH/CMAKE_BUILD_TYPE-build/svf-llvm/extapi.bc !\n";
+    struct stat statbuf;
+    // 1. Set extapi.bc path through setExtBcPath() method
+    if (!extBcPath.empty() && !stat(extBcPath.c_str(), &statbuf))
+        return extBcPath;
+
```

Comment:
You could simplify that function and make it much faster:

1. add a short circuit in the beginning:
```C
if (!extBcPath.empty())
    return extBcPath;
```

2. just assemble the paths via getenv, getFilePath, ... as below but than forward them to `setExtBcPath` and do the stat call there once!

This should also remove the code duplication.